### ISSUE TITLE
Open drawer by clicking toolbar navigation icon

### ISSUE THREAD

**Issue by kboyarshinov** [JakeWharton/u2020#126]
From `TrendingView.java`:
```java
  toolbarView.setNavigationOnClickListener(new OnClickListener() {
      @Override public void onClick(View v) {
        // TODO bind to drawer with... injection?
      }
    });
```

How are you guys planning to implement this binding with `MainActivity.drawerLayout`?

Typically I'm doing this the following way:
`ActivityConnector` binds to Activity livecycle with attach/detach (e.g. called from onStart/onStop), holds reference to some object in Activity. Simply it looks like this:
```java
public class ActivityConnector<T> {
  private WeakReference<T> ref;

  public void attach(T obj) {
    ref = new WeakReference<T>(obj);
  }

  public void detach() {
    ref = null;
  }

  public T get() {
    return ref == null ? null : ref.get();
  }
}
```
More robust implementation can be found [here](https://github.com/LiveTyping/u2020-mvp/blob/master/app/src/main/java/ru/ltst/u2020mvp/base/ActivityConnector.java).

This approach can be used to bind `drawerLayout`:
```java
public final class MainActivity extends Activity {
  @Inject ActivityConnector<DrawerLayout> drawerConnector;

  @InjectView(R.id.main_drawer_layout) DrawerLayout drawerLayout;

  @Override protected void onStart() {
    super.onStart();
    drawerConnector.attach(drawerLayout);
  }

  @Override protected void onStop() {
    super.onStop();
    drawerConnector.detach();
  }
}
```
in `TrendingView.java`:
```java
@Inject ActivityConnector<DrawerLayout> drawerConnector;
@InjectView(R.id.trending_toolbar) Toolbar toolbarView;

@Override protected void onFinishInflate() {
   super.onFinishInflate();
   ButterKnife.inject(this);

   toolbarView.setNavigationIcon(R.drawable.menu_icon);
   toolbarView.setNavigationOnClickListener(new OnClickListener() {
     @Override public void onClick(View v) {
       DrawerLayout drawer = drawerConnector.get();
       if (drawer != null) {
         drawerLayout.openDrawer(Gravity.START);
       }
     }
   });
}
```

What do you think about that?

**Comment by JakeWharton** [comment]
To be honest, I wasn't even going to put an abstract behind it. I was just going to inject the `DrawerLayer` directly.

The `ActivityConnector` indirection isn't needed here because we are a view and already have a reference to the activity in the form of its `Context`.

**Comment by kboyarshinov** [comment]
Thats true. For single activity binding `ActivityConnector` brings no benefit. It was initially created to bind multiple activities and store the top most or current for user. 

What `DrawerLayer` do you mean? 
Doing like `((MainActivity) context).getDrawerLayout().openDrawer(Gravity.START)` seems very coupled code.

**Comment by JakeWharton** [comment]
I meant `DrawerLayout`. We'd add it to an activity-specific object graph so it could simply be `@Inject`ed.

**Comment by kboyarshinov** [comment]
Ahh, ok, I see. I can do that if you don't mind?

**Comment by JakeWharton** [comment]
Sure. We have a semi-clever way of obtaining the injector that I'd like to
introduce. Let me see if I can get it done here quickly.

**Comment by kboyarshinov** [comment]
Ok. I'm interested what this way is. I was playing around several ways of doing this: using custom system service, or even Mortar. In dagger2, however, obtaining specific component is not so obvious, and I'm still looking for more clean way.

**Comment by JakeWharton** [comment]
Heh, I did the system service. I think it's the best for this situation in terms of complexity vs. desired result: https://github.com/JakeWharton/u2020/pull/128.

**Comment by kboyarshinov** [comment]
:+1: fair enough

**Comment by JakeWharton** [comment]
It's merged. Go nuts!

**Comment by kboyarshinov** [comment]
Done. #131
